src/amber/router/params_parser.cr

```diff
@@ -1,28 +1,29 @@
 require "json"
 
 module Amber::Router
-  # The Params module will parse parameters from a URL, a form post or a JSON
+  alias ParamsType = String | JSON::Type
```

Comment:
Should this be JSON::Any?

Reply:
See https://crystal-lang.org/api/0.23.1/JSON/Type.html

Reply:
it will be spending  a lot of time instantiating a JSON::Any for each json param, instead of doing this ahead of time without knowing what params will be use in the request life-cycle the user can determine when he wants to parse out the JSON to the correct type or simply wrap it in a JSON::Any object. JSON::Any is a convenience wrapper to traverse the JSON.

Reply:
But JSON.parse returns JSON::Any not JSON::Type right? So isn't it already doing that work?

Reply:
JSON.parse return JSON::Type See https://github.com/crystal-lang/crystal/blob/e2a1389e8165fb097c785524337d7bb7b550a086/src/json.cr#L83

Reply:
Disregard the my comment above since I was looking at the wrong method 😕

Reply:
Got it thanks.